Packet++/header/TpktLayer.h

```diff
@@ -0,0 +1,122 @@
+#ifndef PACKETPP_TPKT_LAYER
+#define PACKETPP_TPKT_LAYER
+
+#include "EthLayer.h"
+#include "Layer.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * @struct tpkthdr
+	 * Represents a TPKT protocol header
+	 */
+#pragma pack(push, 1)
+	typedef struct
+	{
+		/** message vrsn */
+		uint8_t vrsn;
+		/** message reserved */
+		uint8_t reserved;
+		/** message length */
+		uint16_t length;
+	} tpkthdr;
+#pragma pack(pop)
+
+	/**
+	 * @class TpktLayer
+	 * Represents a TPKT (Transport Service on top of the TCP) protocol layer
+	 */
+	class TpktLayer : public Layer
+	{
+	  public:
+		/**
+		 * Get a pointer to the TPKT header. Data can be retrieved through the
+		 * other methods of this layer. Notice the return value points directly to the data, so every change will change the actual packet data
+		 * @return A pointer to the @ref tpkthdr
+		 */
+		tpkthdr *getTpktHeader() const { return (tpkthdr *)m_Data; }
```

Comment:
This can probably be a private method?

Reply:
If I set it to private, my tests won't run. How do I fix it?

Reply:
In the test it's better to check: 
```cpp
PTF_ASSERT_EQUAL(tpktLayer->getVersion(), 3);
PTF_ASSERT_EQUAL(tpktLayer->getReserved(), 0);
PTF_ASSERT_EQUAL(tpktLayer->getLength(), 607);
```

These methods anyway call `getTpktHeader()` internally, so `getTpktHeader()` can be private